modules/components/pages/outputs/snowflake_streaming.adoc

```diff
@@ -0,0 +1,500 @@
+    mapping: "" # No default (optional)
+    init_statement: | # No default (optional)
+      CREATE TABLE IF NOT EXISTS <table-name> (amount NUMBER);
+    schema_evolution:
+      enabled: false # No default (required)
+      new_column_type_mapping: |-
+        root = match this.value.type() {
+          this == "string" => "STRING"
+          this == "bytes" => "BINARY"
+          this == "number" => "DOUBLE"
+          this == "bool" => "BOOLEAN"
+          this == "timestamp" => "TIMESTAMP"
+          _ => "VARIANT"
+        }
+    build_parallelism: 1
+    batching:
+      count: 0
+      byte_size: 0
+      period: ""
+      check: ""
+      processors: [] # No default (optional)
+    max_in_flight: 64
+    channel_prefix: "" # No default (optional)
+```
+
+--
+======
+
+`<table-name>`: Replace the placeholder value with the name of the Snowflake table you want to write data to.
+
+
+== Supported data formats for Snowflake columns
+
+Your output must create messages with the correct data formats for the Snowflake columns you want to write data to.
+
+|===
+| Snowflake column type | Allowed data formats in Redpanda Connect
+
+| CHAR, VARCHAR 
+| String
+
+| BINARY
+| Byte (`[]byte`)
+
+| NUMBER 
+a| 
+* Any numeric type
+* String representation of a number
+
+| FLOAT 
+| Any numeric type
+
+| BOOLEAN
+a|
+* Boolean (`bool`)
+* Any numeric data type
+* Strings parsable according to https://pkg.go.dev/strconv#ParseBool[`strconv.ParseBool`^] in Golang.
+
+| TIME, DATE, TIMESTAMP 
+| Unix or RFC 3339 with timestamps in nanoseconds. You can parse different string formats using a https://docs.redpanda.com/redpanda-connect/guides/bloblang/about/[Bloblang `mapping`].
+
+| VARIANT, ARRAY, OBJECT
+| Any data type is converted into JSON
+
+| GEOGRAPHY,GEOMETRY
+| Not supported. For full details of all unsupported data types and other limitations, see the https://docs.snowflake.com/en/user-guide/data-load-snowpipe-streaming-overview#limitations[Snowflake documentation^].
+
+|===
+
+
+== Authentication
+
+You can authenticate with Snowflake using an https://docs.snowflake.com/en/user-guide/key-pair-auth[RSA key pair^]. Either specify:
+
+* A PEM-encoded private key, in the <<private_key,`private_key` field>>.
+* The path to a file from which the output can load the private RSA key, in the <<private_key_file,`private_key_file` field>>.
+
+
+== Performance
+
+For improved performance, this output:
+
+* Sends multiple messages in parallel. You can tune the maximum number of in-flight messages (or message batches) with the field `max_in_flight`.
+* Sends messages as a batch. You can configure batches at both the input and output level. For more information, see xref:configuration:batching.adoc[Message Batching].
+
+=== Batch sizes
+
+Redpanda recommends that every message batch writes at least 16 MiB of compressed output to Snowflake. You can monitor batch sizes using the `snowflake_compressed_output_size_bytes` metric.
```

Comment:
Are there more metrics it would be useful to know about? Some components seem to have a list at the start of them.

Reply:
Yes the other metrics are timers: https://github.com/redpanda-data/connect/blob/12b50adab02a54603c374329ea213a290e9550a1/internal/impl/snowflake/output_snowflake_streaming.go#L378-L382

Descriptions

`snowflake_build_output_latency_ns`: The time it takes to build the output file that is sent to snowflake
`snowflake_upload_latency_ns`: The time it takes to upload the output file to snowflake
`snowflake_convert_latency_ns` + `snowflake_serialize_latency_ns` = `snowflake_build_output_latency_ns`
`snowflake_convert_latency_ns`: The time it takes to convert messages into snowflake column data types
`snowflake_serialize_latency_ns`: The time it takes to serialized the converted columnar data into a file to send to snowflake.

Reply:
Metrics table added